uart.cpp

```diff
@@ -158,11 +262,55 @@ uart_input::recall_bindings() {
 
 void uart_input::poll()
 {
+    UartStatus *status = &Application.uart_status;
     auto now = std::chrono::high_resolution_clock::now();
+
+    /* handle baud change */
+    if(status->active_baud_index != status->selected_baud_index) {
+        status->baud = std::stoul(uart_baudrates_list[status->selected_baud_index]);
+        status->active_baud_index = status->selected_baud_index;
+        status->reset_stats();
+        status->is_connected = false;
+        setup_port();
+    }
+
+    /* handle port change */
+    if(status->available_ports.size() > 0 && status->selected_port_index >= 0 && status->active_port_index != status->selected_port_index) {
+        status->port_name = status->available_ports[status->selected_port_index];
+        status->active_port_index = status->selected_port_index;
+        status->reset_stats();
+        status->is_connected = false;
+        setup_port();
+    }
+
+    if (
+        (!port && std::chrono::duration<float>(now - last_port_find).count() > 1.0)
+        || (port && std::chrono::duration<float>(now - last_port_find).count() > 5.0)
```

Comment:
Dlaczego enumeruje porty jak jest podłączony?

Reply:
Realizuje use case, w którym podłączasz dodatkowe urządzenie na które chciałbyś się przepiąć. Bez tego trzeba byłoby wyłączać komunikację. Średnio mi się to podoba (stąd rzadszy polling) i wolałbym to mieć na eventach, tyle że wtedy dochodzi zrobienie dedykowanego wątku nasłuchującego. Nie bardzo chciałem wchodzić w threading, ale jak trzeba to rozgryzę temat.

Reply:
myślę że z guzikiem będzie ok